gammapy/irf/tests/test_background.py

```diff
@@ -49,3 +53,175 @@ def test_background_3d_write(bkg_3d):
     hdu = table_to_fits_table(bkg_3d.to_table())
     assert_equal(hdu.data['DETX_LO'][0], bkg_3d.data.axis('detx').lo.value)
     assert hdu.header['TUNIT1'] == bkg_3d.data.axis('detx').lo.unit
+
+
+def make_test_array():
+    # Create a dummy `Background2D`
+    # Make Axes
+    nE_bins = 100
+    ebounds = EnergyBounds.equal_log_spacing(0.1, 100, 100, 'TeV')
+    offset = Angle(np.linspace(0, 2.5, 100), "deg")
+    data = np.zeros((nE_bins, len(offset) - 1))
+    # Make dummy data
+    data[2, 23] = 1
+    data[78, 59] = 1
+    data_unit = u.Unit('s-1 MeV-1 sr-1')
+    bkg_2d = Background2D(energy_lo=ebounds[0:-1], energy_hi=ebounds[1:],
+                          offset_lo=offset[0:-1], offset_hi=offset[1:],
+                          data=data * data_unit)
+    return bkg_2d
+
+
+def make_test_cube():
+    # Create a dummy `Background2D`
+    # Make Axes
+    nE_bins = 100
+    ebounds = EnergyBounds.equal_log_spacing(0.1, 100, nE_bins, 'TeV')
+    offset = Angle(np.linspace(0, 2.5, 100), "deg")
+    bins = 2 * len(offset)
+    coordx_edges = Angle(np.linspace(-offset.max().value, offset.max().value, bins + 1), "deg")
+    coordy_edges = Angle(np.linspace(-offset.max().value, offset.max().value, bins + 1), "deg")
+    data = np.zeros((nE_bins, bins, bins))
+    # Make dummy data
+    data[3, 10, 152] = 1
+    data[20, 108, 52] = 1
+    data_unit = u.Unit('s-1 MeV-1 sr-1')
+    bkg_3d = Background3D(energy_lo=ebounds[0:-1], energy_hi=ebounds[1:], detx_lo=coordx_edges[0:-1],
+                          detx_hi=coordx_edges[1:], dety_lo=coordy_edges[0:-1], dety_hi=coordy_edges[1:],
+                          data=data * data_unit)
+    return bkg_3d
+
+
+def test_background2d_read_write(tmpdir):
+    bkg_2d_1 = make_test_array()
+    # hdu_list=bkg_2d_1.to_hdulist(name='BACKGROUND')
```

Comment:
@cdeil I wanted to use directly ```to_hdulis()```method not to have to write anything on the disk but I had some issues for the unit when I used from_hdulist(). Here is an example in the method of the output I get:
```
In [18]:     bkg_2d=Background2D.read(tmpdir+"/bkg2d.fits")
    ...:     hdu_list=bkg_2d_1.to_hdulist(name='BACKGROUND')
    ...:     bkg_2d_fromhdu=Background2D.from_hdulist(hdulist=hdu_list, hdu='BAC
    ...: KGROUND')
    ...: 

In [19]: axis = bkg_2d.data.axis('energy')

In [20]: axis.unit
Out[20]: Unit("TeV")

In [21]: axis_fromhdu = bkg_2d_fromhdu.data.axis('energy')

In [22]: axis_fromhdu.unit
Out[22]: Unit(dimensionless)
```

So I don't know why I get unit without dimension using directly fom_hdulist().
I can imagine it comes from the line, `fits.open ` used in the `read`method since with the read method I get a numpy.record as one of the column of my hdu data (you can see it with that line ```fits.open(tmpdir+"/bkg2d.fits"```)[1].data) and I don't get this column with directly the data in the hdulist ```hdu_list[1].data```